engine/src/org/pentaho/di/trans/ael/adapters/TransMetaConverter.java

```diff
@@ -32,18 +32,20 @@
 import org.pentaho.di.trans.TransMeta;
 import org.pentaho.di.trans.step.StepMeta;
 
+import java.util.UUID;
 import java.util.function.Consumer;
 import java.util.stream.IntStream;
 
 public class TransMetaConverter {
 
   public static final String TRANS_META_CONF_KEY = "TransMeta";
+  public static final String TRANS_META_NAME_CONF_KEY = "TransMetaName";
   public static final String STEP_META_CONF_KEY = "StepMeta";
 
 
   public static Transformation convert( TransMeta transMeta ) {
     final org.pentaho.di.engine.model.Transformation transformation =
-      new org.pentaho.di.engine.model.Transformation( transMeta.getName() );
+      new org.pentaho.di.engine.model.Transformation( UUID.randomUUID().toString() );
```

Comment:
Thinking we should probably keep transMeta.getName(), but append a string to make it unique.  That way logging can provide some idea of what the source transformation is.